Library/brew.sh

```diff
@@ -207,6 +173,56 @@ EOS
   esac
 fi
 
+if [[ -z "$HOMEBREW_DEVELOPER" ]]
+then
+  unset HOMEBREW_RUBY_PATH
+fi
+
+if [[ -z "$HOMEBREW_RUBY_PATH" && "$HOMEBREW_COMMAND" != "install-vendor" ]]
+then
+  HOMEBREW_VENDOR_RUBY_PATH="$HOMEBREW_LIBRARY/Homebrew/vendor/ruby/opt/bin/ruby"
+  if [[ -x "$HOMEBREW_VENDOR_RUBY_PATH" ]]
+  then
+    HOMEBREW_RUBY_PATH="$HOMEBREW_VENDOR_RUBY_PATH"
+  else
+    if [[ -n "$HOMEBREW_OSX" ]]
+    then
+      HOMEBREW_RUBY_PATH="/System/Library/Frameworks/Ruby.framework/Versions/Current/usr/bin/ruby"
+    else
+      HOMEBREW_RUBY_PATH="$(which ruby)"
+    fi
+
+    if [[ -z "$HOMEBREW_RUBY_PATH" || "$("$HOMEBREW_RUBY_PATH" -e "puts RUBY_VERSION.split('.').first")" = "1" ]]
```

Comment:
I think we should be stricter about the Ruby version here. We should check for at least Ruby 2 (not `0` or `3`) and consider doing e.g. Ruby 2.0.0 if we hit backwards compatibility issues.